InstantPrintTool.py

```diff
@@ -239,6 +243,24 @@ def __export(self):
         if not success:
             QMessageBox.warning(self.iface.mainWindow(), self.tr("Print Failed"), self.tr("Failed to print the composition."))
 
+    def __print(self):
+        self.printer = QPrinter()
```

Comment:
if not self.printer:
        self.printer = QPrinter()

(You save some time by reusing the existing printer object on subsequent calls to print)